pkg/operator/deploy/deploy.go

```diff
@@ -157,9 +164,10 @@ func (o *operator) resources() ([]runtime.Object, error) {
 				Name: arov1alpha1.SingletonClusterName,
 			},
 			Spec: arov1alpha1.ClusterSpec{
-				ResourceID: o.oc.ID,
-				ACRDomain:  o.env.ACRDomain(),
-				Location:   o.env.Location(),
+				ResourceID:    o.oc.ID,
+				ACRDomain:     o.env.ACRDomain(),
+				AZEnvironment: o.env.Environment().Name,
+				Location:      o.env.Location(),
```

Comment:
you are not setting vnet from above

Reply:
It's below the `GenevaLogging`.  Shown [on line 183](https://github.com/Azure/ARO-RP/pull/1230/files#diff-807d06e3678eec1759c7bc5bf8b786fe7bcce925762e9a7db87c3c84c4bb103bR183), right above credentialsRequest